lib/orphans.nix

```
@@ -0,0 +1,104 @@
+{ lib }:
+let
+  inherit (lib)
+    types
+    optional
+    isDerivation
+    attrByPath
+    flatten
+    any
+    assertMsg
+  ;
+  inherit (builtins)
+    length
+    concatStringsSep
+    mapAttrs
+    attrValues
+    tryEval
+  ;
+  isAttr = types.attrs.check;
+in
+rec {
+  /* Return a list of maintainers from a package, empty list if inaccessible.
+
+     Example:
+       listMaintainers pkgs.foo
+       => [ { ... } ]
+       listMaintainers pkgs.bar
+       => [ ]
+       listMaintainers {}
+       => error: listMaintainers must receive a package
+  */
+  listMaintainers = p:
+  if isDerivation p then
+    (attrByPath [ "meta" "maintainers" ] [ ] p)
```

```suggestion
    p.meta.maintainers or []
```

Slightly faster as well, fwiw.